app/controllers/category_controller.rb

```diff
@@ -0,0 +1,20 @@
+class CategoryController < ApplicationController
+  def index
+    categories = Category.all
+
+    render locals: {categories: categories}
+  end
+
+  def show
+    if params[:id].present?
+      category = Category.find(params[:id])
+    end
```

Comment:
If the `id` isn't present you'll still try and get the posts on line `13` and run into an `undefined method on nil` error.

What do you think about using the `#find!` method instead here to raise an exception?